File: output/ChatGPT/codenet/Python/C++/s957164965.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <map>

using namespace std;

int main() {
    int X, N;
    cin >> X >> N;
    
    vector<int> p(N);
    for (int i = 0; i < N; i++) {
        cin >> p[i];
    }
    
    sort(p.begin(), p.end());
    
    map<int, vector<int>> candidates;
    for (int i = -1; i <= 101; i++) {
        if (find(p.begin(), p.end(), i) != p.end()) {
            continue;
        }
        
        if (candidates.find(abs(X-i)) == candidates.end()) {
            candidates[abs(X-i)] = vector<int>(1, i);
        } else {
            candidates[abs(X-i)].push_back(i);
        }
    }
    
    int smallest = min_element(candidates.begin(), candidates.end())->first;
    
    int minNum = *min_element(candidates[smallest].begin(), candidates[smallest].end());
    
    cout << minNum << endl;

    return 0;
}

File: output/ChatGPT/codenet/Python/Java/s957164965.java
import java.util.*;

public class s957164965 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int X = scanner.nextInt();
        int N = scanner.nextInt();
        List<Integer> p = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            p.add(scanner.nextInt());
        }
        
        Collections.sort(p);
        
        Map<Integer, List<Integer>> candidates = new HashMap<>();
        for (int i = -1; i <= 101; i++) {
            if (p.contains(i)) {
                continue;
            }
            
            int diff = Math.abs(X - i);
            if (!candidates.containsKey(diff)) {
                candidates.put(diff, new ArrayList<>());
            }
            candidates.get(diff).add(i);
        }
        
        int smallest = Collections.min(candidates.keySet());
        int minNum = Collections.min(candidates.get(smallest));
        
        System.out.println(minNum);
    }
}